tests/multi_syscall_test.rs

```diff
@@ -0,0 +1,288 @@
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.retdata, vec![caller_address.clone().0])
+    }
+
+    // Block for get_contact_address.
+    {
+        let call_info = test_syscall(
+            "contract_address",
+            address.clone(),
+            vec![],
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.events, vec![])
+    }
+    // Block for get_execution_info_syscall.
+    {
+        let call_info = test_syscall(
+            "execution_info_syscall",
+            address.clone(),
+            calldata.clone(),
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.events, vec![]);
+    }
+
+    // Block for replace_class_syscall
+    {
+        let call_info = test_syscall(
+            "replace_class_syscall_test",
+            address.clone(),
+            calldata.clone(),
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.events, vec![])
+    }
+
+    // Block for library_call_syscall
+    {
+        let entrypoint_selector =
+            Felt252::from_bytes_be(&calculate_sn_keccak("get_number".as_bytes()));
+        let new_call_data = vec![
+            Felt252::from_bytes_be(&class_hash),
+            entrypoint_selector,
+            Felt252::from(25),
+        ];
+        let call_info = test_syscall(
+            "test_library_call_syscall",
+            address.clone(),
+            new_call_data,
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.retdata, vec![25.into()])
+    }
+
+    // Block for call_contract_syscall
+    {
+        let entrypoint_selector =
+            Felt252::from_bytes_be(&calculate_sn_keccak("get_number".as_bytes()));
+        let new_call_data = vec![entrypoint_selector, Felt252::from(25)];
+        let call_info = test_syscall(
+            "test_call_contract_syscall",
+            address.clone(),
+            new_call_data,
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.retdata, vec![25.into()])
+    }
+
+    // Block for send_message_to_l1_syscall
+    {
+        let new_call_data = vec![2222.into(), Felt252::from(25), Felt252::from(30)];
+        let call_info = test_syscall(
+            "test_send_message_to_l1",
+            address.clone(),
+            new_call_data,
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.retdata, vec![])
```

Comment:
Please test that the messages are sent correctly. You can get the messages using: `call_info.l2_to_l1_messages`